__tests__/components/DurationInput.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { ValidationProvider } from 'vee-validate';
+import VueI18n from 'vue-i18n';
+import Vuex from 'vuex';
+
+StandardValidationRules.register();
+
+interface DurationInputProps {
+    value?: string;
+    targetAsset?: 'mosaic' | 'namespace';
+    label?: string;
+    showRelativeTime?: boolean;
+}
+
+describe('components/DurationInput', () => {
+    const networkConfiguration = { blockGenerationTargetTime: 30 };
+    const mockNetworkStore = {
+        namespaced: true,
+        state: { networkConfiguration },
+        getters: {
+            networkConfiguration: (state) => {
+                return state.networkConfiguration;
+            },
+        },
+    };
+    const getDurationInputWrapper = (props?: DurationInputProps) => {
+        const localVue = createLocalVue();
+        localVue.component('ValidationProvider', ValidationProvider);
+        localVue.use(Vuex);
+        localVue.use(VueI18n);
+        const store = new Vuex.Store({
+            modules: {
+                network: mockNetworkStore,
+            },
+        });
+        const options = {
+            localVue,
+            i18n,
+            store,
+            propsData: { ...props },
+            stubs: ['Tooltip'],
+        };
+        const wrapper = mount(DurationInput, options);
+        return wrapper;
+    };
+
+    test('renders component', () => {
+        // Arrange + Act:
+        const wrapper = getDurationInputWrapper();
+
+        // Assert:
+        expect(wrapper.find('input[type="number"]').exists()).toBe(true);
+    });
+
+    test('relative time is unlimited when no value is set', () => {
+        // Arrange + Act:
+        const wrapper = getDurationInputWrapper({ showRelativeTime: true });
+
+        // Assert:
+        expect(wrapper.find('span.relative-time').text()).toContain(i18n.t('label_duration_unlimited'));
+    });
+
+    test('relative time is unlimited when value is zero', async () => {
+        // Arrange:
+        const wrapper = getDurationInputWrapper({ showRelativeTime: true, value: '0' });
+
+        // Assert:
+        expect(wrapper.find('span.relative-time').text()).toContain(i18n.t('label_duration_unlimited'));
+    });
+
+    test('relative time is presented when value is valid', async () => {
+        // Arrange + Act:
+        const wrapper = getDurationInputWrapper({ showRelativeTime: true, value: '10000' });
+
+        // Assert:
+        expect(wrapper.find('span.relative-time').text()).toContain('3 d 11 h 20 m');
+    });
+
+    test('emits input event when chosen value is set', async () => {
+        // Arrange:
+        const chosenValue = '20000';
+        const wrapper = getDurationInputWrapper({ showRelativeTime: true, targetAsset: 'namespace', value: '10000' });
+        const component = wrapper.vm as DurationInputTs;
+        component.chosenValue = chosenValue;
+
+        // Assert:
+        expect(wrapper.emitted('input')[0][0]).toBe(chosenValue);
+    });
+});
```

Comment:
should you check validationRule somehow?

Reply:
I will include it to the next PR 👍